Makefile

```diff
@@ -38,6 +38,10 @@ linker_flags = -s -w -extldflags "-fno-PIC -static" -X main.version=$(VERSION) -
 build: build-cli ## Build all Go binaries
 
 build-cli: output-dir ## Build the nginx-meshctl binary
+	@## check if version is set
+	@if [ -z "$(VERSION)" ]; then \
+	  echo '$@:WARNING: building the CLI with an empty version will break upgrade capability'; \
+	fi
```

Comment:
The only way the version string would be empty is if a dev explicitly sets it to be empty, which I would not expect someone to intentionally do. The line at the top of this file `VERSION ?= $(shell git describe --tags)` ensures that it's set by default. So I don't expect this check to work in the general case.

However, the default value doesn't allow upgrades to work since VERSION has to be set to a valid release tag. Until we add support for the `--imageTag` flag via the upgrade command, a dev will have to manually set the VERSION to the release tag when building.

I think a better fix here is improve the error that the CLI spits out if the upgrade command is called and `version` is not a valid semver.